cognite/client/_api/functions.py

```diff
@@ -270,54 +291,55 @@ def create(
                 >>> function = client.functions.create(name="myfunction", function_handle=handle)
 
             .. note::
-                When using a predefined function object, you can list dependencies between the tags `[requirements]` and `[/requirements]` in the function's docstring. The dependencies will be parsed and validated in accordance with requirement format specified in `PEP 508 <https://peps.python.org/pep-0508/>`_.
+                When using a predefined function object, you can list dependencies between the tags `[requirements]` and `[/requirements]` in the function's docstring.
+                The dependencies will be parsed and validated in accordance with requirement format specified in `PEP 508 <https://peps.python.org/pep-0508/>`_.
         '''
-        self._assert_exactly_one_of_folder_or_file_id_or_function_handle(folder, file_id, function_handle)
-
-        if folder:
-            validate_function_folder(folder, function_path, skip_folder_validation)
-            file_id = self._zip_and_upload_folder(folder, name, external_id, data_set_id)
-        elif function_handle:
-            _validate_function_handle(function_handle)
-            file_id = self._zip_and_upload_handle(function_handle, name, external_id, data_set_id)
-        assert_type(cpu, "cpu", [Number], allow_none=True)
-        assert_type(memory, "memory", [Number], allow_none=True)
-
-        sleep_time = 1.0  # seconds
-        for i in range(MAX_RETRIES):
-            file = self._cognite_client.files.retrieve(id=file_id)
-            if file and file.uploaded:
-                break
-            time.sleep(sleep_time)
-            sleep_time *= 2
-        else:
-            raise OSError("Could not retrieve file from files API")
+        if not exactly_one_is_not_none(function, name):
+            raise TypeError("Exactly one of name or function must be specified.")
```

Comment:
This error might seem a bit "random" to someone not familiar with the two options of how to create a function, here's a suggested wording:

```suggestion
            raise ValueError(
                "Either pass a FunctionWrite object to be created via the 'function' argument, or pass all "
                "individual arguments (in which case 'name' is required)."
            )
```